### PR TITLE
Submitting HW#11 (Blog) w/ Comments

### PR DESCRIPTION
**Pull request by snarvasa**
Let me know if you have any issues with this PR.

giphy

### CODE REVIEW

File: SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class ArticlesController < ApplicationController
+
+  http_basic_authenticate_with name: "dhh", password: "secret", except: [:index, :show] #authenticates dhh on every action, below
+
```
Comment (chylaudes) [inline SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb]: To keep yourself DRY, you can create a private method called `set_article` and use the `before_action`
```
  before_action :set_article, only: [:show, :edit, :update, :destroy]
```
the before_action tells the Articles controller to use the :get_article method on the show, edit, update, and destroy page.  So you do not need `@article = Article.find(params[:id])`

File: SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @article = Article.find(params[:id])            #GETS the article ID and stores it in @article
+    @article.destroy                                #destroys the article
+
+    redirect_to articles_path                       #redirects the user to the articles index
+  end
+
+private
+  def article_params
+    params.require(:article).permit(:title, :text)  #makes title + text privately available for any method requiring article params
+  end
```
Comment (chylaudes) [inline SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb]: you can define the `set_article method here`
```
def set_article 
 @article = Article.find(params[:id])
end
```

File: SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  def update                                        #Method to update Article
+    @article = Article.find(params[:id])            #GETs the article ID, assigns it to @article
+
+    if @article.update(article_params)              #if the article is updated
+      redirect_to @article                          #redirect to the article show page
+    else                                            #otherwise
+      render 'edit'                                 #render the Edit page
+    end
+  end
+end
```
Comment (chylaudes) [inline SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb]: delete the `end ` on line 40

File: SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb (hunk 2 above)
Comment (chylaudes) [inline SeanNarvasa-snarvasa/hw_11/blog/app/controllers/articles_controller.rb]: add an 
`end`
here

**Comment by chylaudes** [comment]
:+1: :+1: :+1: :+1: :+1:  
giphy


Great!  I just added a couple of tips to keep your Code dry and corrected something on your articles_controller.  I liked how you're using font-awesome, and bootstrap and incorporating javascript into your web page!  hahaha

**Comment by snarvasa** [comment]
@chylaudes Nice! Thanks for the suggestions.

**Pull request merged**